### PR TITLE
Fixes error binding context in layout

### PR DESCRIPTION
**Pull request by Magentaize**

### CODE REVIEW

File: source/Handlebars/Compiler/Structure/BindingContext.cs [hunk 1]
```diff
@@ -87,6 +87,10 @@ private object GetContextVariable(string variableName, object target)
         {
             object returnValue;
             variableName = variableName.TrimStart('@');
+            if (target.GetType() == typeof(HandlebarsCompiler.DynamicViewModel))
+            {
+                target = ((HandlebarsCompiler.DynamicViewModel) target).Objects[1];
```
[the remark this answers is not in the document]
Comment (Magentaize) [reply source/Handlebars/Compiler/Structure/BindingContext.cs]: At HandlebarsCompiler.cs line 64, it passes a  DynamicViewModel instance with a argument `object[]` which the second object is the corresponding view model

[the remark this answers is not in the document]
Comment (rexm) [reply source/Handlebars/Compiler/Structure/BindingContext.cs]: Positional significance is a code smell. If specific elements in the array have unique meaning, they should be made part of the model.

[the remark this answers is not in the document]
Comment (Magentaize) [reply source/Handlebars/Compiler/Structure/BindingContext.cs]: You're right, it's a bad design and the whole #76 should be re-considered.

**Comment by rexm** [comment]
Going to avoid any additional investment in the baked-in view engine and focus on a better extensibility model in [v2](https://github.com/rexm/Handlebars.Net/issues/294) that will afford well-designed view engines.

**Pull request closed without merging**